### PR TITLE
spherical harmonic wrapper

### PR DESCRIPTION
**Pull request by camelto2**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

I use the Ylm's in Numerics for the spin-orbit ECP, and will need cartesian derivatives of these for ionic derivatives. The Ylm's in Numerics require cartesian vectors as input...except they **must**  unit vectors and must be in the format [z,x,y] instead of the standard ordering. 

This just adds a sphericalHarmonic wrapper which takes in any cartesian vector so the user can simply provide whatever position they want. Internally, it calculates the unit vector and reorders it to be [z,x,y] and then calls the Ylm implementation we already have. This allows me to clean up the SOECP for better readability.

I also introduce a sphericalHarmonicGrad, which calculates (d/dx, d/dy, d/dz)Ylm. This will be need for ionic derivatives of the SOECP.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
my mac

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/Numerics/Ylm.h [hunk 1]
```diff
@@ -127,5 +127,44 @@ inline void derivYlmSpherical(const int l,
   }
 }
 
+// wrapper for Ylm, which doesn't require a unit vector and is in the normal order (x,y,z)
```
Comment (ye-luo) [inline src/Numerics/Ylm.h:130]: In the function which needs (x,y,z) ordering, please add a note in the function documentation.
Here specify the ordering needed by this function.

**Review by rcclay — APPROVED** [review]
LGTM

File: src/Numerics/Ylm.h (hunk 1 above)
Comment (camelto2) [reply src/Numerics/Ylm.h]: added and updated comments. Should be more clear

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Comment by prckent** [comment]
Thanks for maintaining the high test coverage in Ylm.h

**Pull request merged**